app/server-utils/kv-cache.ts

```diff
@@ -1,16 +1,18 @@
 import type {DataFunctionArgs} from '@remix-run/cloudflare'
-
+type RequestForReload = DataFunctionArgs['request'] | 'NEVER_RELOAD'
 export function withCache<Fn extends (...args: string[]) => Promise<any>>(
   defaultKey: string,
   fn: Fn
 ): (
-  request?: DataFunctionArgs['request'],
+  // pass the real Request object when possible, 'NEVER_RELOAD' is an escape hatch for nested withCache(),
+  // it's used for detection of `?reload` in url, to invalidate cache in a background request
+  request: RequestForReload,
   ...args: Parameters<Fn>
 ) => Promise<{data: Awaited<ReturnType<Fn>>; timestamp: string}> {
-  return (async (request?: DataFunctionArgs['request'], ...args: Parameters<Fn>) => {
+  return (async (request: RequestForReload, ...args: Parameters<Fn>) => {
     const key = args[0] ?? defaultKey
 
-    const shouldReload = request?.url.match(/[?&]reload/)
+    const shouldReload = request === 'NEVER_RELOAD' ? false : request.url.match(/[?&]reload/)
```

Comment:
the bug was caused by forgetting to pass a request parameter, so we cannot make it an optional paramater to make sure to avoid the same hard to debug cache invalidation bugs that survived unfixed for a few months..

if you can remove the nested withCache usage for glossary, where we don't have access to the real request, feel free to also remove the string variant from TS union

I believe TS unions are actually the right solution for this kind of problem when we want to make some parameter mandatory, but it's not possible to have it everywhere and need an escape hatch, but really really want to discourage usage of the escape hatch